internal/dag/cache_test.go

```diff
@@ -2074,6 +2088,22 @@ func TestServiceTriggersRebuild(t *testing.T) {
 			svc:  service("default", "service-1"),
 			want: false,
 		},
+		"grpcroute exists in same namespace as service": {
+			cache: cache(
+				service("default", "service-1"),
+				grpcRoute("default", "service-1"),
+			),
+			svc:  service("default", "service-1"),
+			want: true,
+		},
+		"grpcroute does not exist in same namespace as service": {
+			cache: cache(
+				service("default", "service-1"),
+				grpcRoute("user", "service-1"),
+			),
+			svc:  service("default", "service-1"),
+			want: false,
+		},
```

Comment:
Please add one more test case, "grpcroute does not use same name as service" (looks like we're missing it for HTTPRoute but it exists for TLSRoute and TCPRoute). Should be a GRPCRoute that references a _different_ service than the one that changed, and the expected result is `false`.

Reply:
Thanks for quick review.
Add a test case.
Rebased.